src/rules/fileHeaderRule.ts

```diff
@@ -37,19 +54,49 @@ export class Rule extends Lint.Rules.AbstractRule {
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
         const { text } = sourceFile;
+        const headerFormat = new RegExp(this.ruleArguments[0] as string);
+        const textToInsert = this.ruleArguments[1] as string | undefined;
+
         // ignore shebang if it exists
         let offset = text.startsWith("#!") ? text.indexOf("\n") : 0;
-        // returns the text of the first comment or undefined
-        const commentText = ts.forEachLeadingCommentRange(
+        // returns details about the first comment or undefined
+        const commentDetails  = ts.forEachLeadingCommentRange(
             text,
             offset,
-            (pos, end, kind) => text.substring(pos + 2, kind === ts.SyntaxKind.SingleLineCommentTrivia ? end : end - 2));
-        if (commentText === undefined || !new RegExp(this.ruleArguments[0] as string).test(commentText)) {
-            if (offset !== 0) {
-                ++offset; // show warning in next line after shebang
+            (pos, end, kind) => ({pos, end, kind}));
+
+        if (offset !== 0) {
+            ++offset; // show warning in next line after shebang
+        }
+
+        if (commentDetails === undefined) {
+            const fix = textToInsert !== undefined
+                ? Lint.Replacement.appendText(offset, this.createComment(sourceFile, textToInsert))
+                : undefined;
+            return [new Lint.RuleFailure(sourceFile, offset, offset, Rule.FAILURE_STRING, this.ruleName, fix)];
+        } else {
+            const { pos, end, kind } = commentDetails;
+            const commentText = text.substring(pos, kind === ts.SyntaxKind.SingleLineCommentTrivia ? end : end - 2);
+            if (!headerFormat.test(commentText)) {
+                const isFirstCommentAHeader = ONLY_WHITESPACE.test(text.substring(offset, pos));
+                const fix = textToInsert !== undefined
+                    ? isFirstCommentAHeader
+                        ? Lint.Replacement.replaceFromTo(pos, end, this.createComment(sourceFile, textToInsert, 0))
```

Comment:
Is removing the existing comment really expected? It could contain JSDoc or something similar useful.

Reply:
Hmm, it's a tricky call. If it's a malformed header, you'd want it to be removed, otherwise you probably wouldn't want it to be. I don't think there's any way to automatically determine which is the case, so we'll want to just pick the option that is most correct in most cases.

I think I agree with your thoughts that replacing the header might be unexpected more often than desired. (Plus it'll be simpler code w/o that feature.) I guess we could make the behavior configurable in the future if desired